src/app/plugins/dog.plugin.ts

```diff
@@ -26,21 +27,25 @@ export class DogPlugin extends Plugin {
       .get(`${this._API_URL}breeds/list/all`)
       .then((response: IHttpResponse) => {
         const breedData = response.data.message;
-        this._breeds = this._allBreeds = Object.keys(breedData);
+
+        this._breeds = Object.keys(breedData);
+        this._allBreeds = new Set(Object.keys(breedData));
 
         // The json is annoyingly {<breed>: [<subBreeds>]} apposed to {breed: <breed>, subBreed: [<subBreeds>]}} so this gets that
-        this._breeds.forEach((breed: string) => {
+        this._subBreeds = this._breeds.flatMap((breed: string) => {
```

Comment:
Never heard of a flatMap, but i kinda see what you are doing.
But returning an empty array | IDogBreed doesnt really make sense.
Just filter out the ones with no subbreeds then map
`this._breeds.filter( > 0 subreeds).map(...)`

Reply:
the empty arrays get flattened out. Since it doesn't accept no return.

Reply:
oh wait I just remebered I didn't need that flat Map that was for something else